runtime/compiler/control/CompilationThread.cpp

```diff
@@ -8787,13 +8788,20 @@ TR::CompilationInfoPerThreadBase::compile(
                {
                TR_ASSERT(_compiler.getHotnessName(_compiler.getMethodHotness()), "expected to have a hotness string");
                if (_compiler.getOutFile() != NULL && _compiler.getOption(TR_TraceAll))
-                  traceMsg(&_compiler, "<compile\n"
-                          "\tmethod=\"%s\"\n"
-                          "\thotness=\"%s\"\n"
-                          "\tisProfilingCompile=%d>\n",
-                          _compiler.signature(),
-                          _compiler.getHotnessName(_compiler.getMethodHotness()),
-                          _compiler.isProfilingCompilation());
+                  {
+                  traceMsg(&_compiler, "<compile\n");
+                  traceMsg(&_compiler, "\tmethod=\"%s\"\n", _compiler.signature());
+                  traceMsg(&_compiler, "\thotness=\"%s\"\n", _compiler.getHotnessName(_compiler.getMethodHotness()));
+                  traceMsg(&_compiler, "\tisProfilingCompile=%d", _compiler.isProfilingCompilation());
+#if defined(JITSERVER_SUPPORT)
+                  if (_compiler.isOutOfProcessCompilation() && TR::compInfoPT->getClientData()) // using jitserver && client JVM
+                     {
+                     traceMsg(&_compiler, "\n");
+                     traceMsg(&_compiler, "\tclientID=" PRIu64 "", TR::compInfoPT->getClientData()->getClientUID());
```

Comment:
Concatenating the empty string isn't helpful; the format string should be just `"\tclientID=" PRIu64` (likewise in `javadump.cpp`).

Reply:
I was confused by the usage of `PRIu64` earlier. The intension of this piece of code is to print the value of `clientUID` instead of the string literal `PRlu64` (`unsigned long long`). 

I tried `"\tclientID=" PRIu64` to write format string in `CompilationThread.cpp`
``` c++
traceMsg(&_compiler, "\tclientID=" PRIu64, TR::compInfoPT->getClientData()->getClientUID());
```
and got below compilation trace
```
clientID=lu
```

However if I use `"\tclientID=%" PRIu64 ""`
``` c++
traceMsg(&_compiler, "\tclientID=%" PRIu64 "", TR::compInfoPT->getClientData()->getClientUID());
```
compilation trace would be
```
clientID=2145050389>
```

Reply:
Similar in `javadump.cpp`, I am using the following line to generate javacore file
``` c++
_OutputStream.writeInteger64(jitConfig->clientUID, PRIu64);
```
Javacore
```
1CICLIENTID    Client UID lu
```
If I use the following
``` c++
_OutputStream.writeInteger64(jitConfig->clientUID, "%" PRIu64 "");
```
Javacore
```
1CICLIENTID    Client UID 3068314397
```

Reply:
How about:
`_OutputStream.writeInteger64(jitConfig->clientUID, "%" PRIu64);`

Reply:
Confirmed. Pushed the latest change into this PR.

Reply:
Sorry, I omitted the `%` in my suggestion: it should have been:
`"\tclientID=%" PRIu64`

Reply:
@keithc-ca Thank you, this has been fixed in the latest change. Please take a look.